graph/src/data/subgraph/mod.rs

```diff
@@ -420,16 +425,17 @@ impl UnresolvedMapping {
 }
 
 #[derive(Clone, Debug, Hash, Eq, PartialEq, Deserialize)]
-pub struct BaseDataSource<M> {
+pub struct BaseDataSource<M, T> {
     pub kind: String,
     pub network: Option<String>,
     pub name: String,
     pub source: Source,
     pub mapping: M,
+    pub templates: Option<Vec<T>>,
 }
 
-pub type UnresolvedDataSource = BaseDataSource<UnresolvedMapping>;
-pub type DataSource = BaseDataSource<Mapping>;
+pub type UnresolvedDataSource = BaseDataSource<UnresolvedMapping, UnresolvedDataSourceTemplate>;
+pub type DataSource = BaseDataSource<Mapping, DataSourceTemplate>;
```

Comment:
If `UnresolvedMapping` always goes with `UnresolvedDataSourceTemplate`, and `Mapping` always goes with `DataSourceTemplate`, wouldn't it be better if the `templates` lived on the `Mapping`?